src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,359 @@
+
+        log.info("init cluster capacity info");
+        maxStandardInstancesInCluster = FleetshardUtils.getClusterCapacityFromMKAgent(oc, ManagedKafkaType.standard);
+        log.debug("Max standard instances according to MKAgent capacity '{}'", maxStandardInstancesInCluster);
+        maxDeveloperInstancesInCluster = FleetshardUtils.getClusterCapacityFromMKAgent(oc, ManagedKafkaType.standard);
+        log.debug("Max developer instances according to MKAgent capacity '{}'", maxDeveloperInstancesInCluster);
+
+        log.info("init kafka management");
+        var apps = ApplicationServicesApi.applicationServicesApi(
+            Environment.PRIMARY_USERNAME,
+            Environment.PRIMARY_PASSWORD);
+        kafkaMgmtApi = apps.kafkaMgmt();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown() {
+
+        // delete kafka instance
+        try {
+            KafkaMgmtApiUtils.cleanKafkaInstance(kafkaMgmtApi, KAFKA_INSTANCE_NAME);
+        } catch (Throwable t) {
+            log.error("clean main kafka instance error: ", t);
+        }
+    }
+
+    @DataProvider(name = "managedKafkaTypes")
+    public Object[][] managedKafkaTypeDataProvider() {
+        return new Object[][]{
+            {"standard"},
+            {"developer"}};
+    }
+
+    @Test(dataProvider = "managedKafkaTypes")
+    @SneakyThrows
+    public void testReservedDeploymentExistence(String mkType) {
+
+        String namespace = String.format("reserved-kafka-%s-1", mkType);
+        log.info("evaluating namespace '{}'", namespace);
+
+        String managedKafkaName = String.format("reserved-kafka-%s-1", mkType);
+        log.info("managed kafka reserved deployment '{}'", managedKafkaName);
+        Optional<ManagedKafka> mkOptional = FleetshardUtils.managedKafka(oc).inNamespace(namespace).list().getItems().stream().filter(ManagedKafka::isReserveDeployment).findAny();
+        Assert.assertTrue(mkOptional.isPresent(), "reserved deployment is not present");
+    }
+
+    @Test(dataProvider = "managedKafkaTypes")
+    public void testReservedDeploymentDoesNotPreventKafkaCreation(String mkType) throws Exception {
+
+        log.info("testing managed kafka type: '{}'", mkType);
+
+        // obtain max limit of kafka instances per given mk type (developer, standard)
+        int upperLimitPerManagedKafkaType;
+        switch (mkType) {
+            case "standard":
+                upperLimitPerManagedKafkaType = maxStandardInstancesInCluster;
+                break;
+            case "developer":
+                upperLimitPerManagedKafkaType = maxDeveloperInstancesInCluster;
+                break;
+            default:
+                throw new Exception("Unsupported managed kafka type");
+        }
+        log.info("upper limit for instance type '{}' is '{}' instances.", mkType, upperLimitPerManagedKafkaType);
+
+        int currentInstancesCount = FleetshardUtils.getCountOfExistingGivenManagedKafkaCRType(oc, ManagedKafkaType.valueOf(mkType));
```

Comment:
The `getCountOfExistingGivenManagedKafkaCRType` returns the total count of MK CRs - `reserved` and `real` deployments. I think what you are interested in here is the number of `real` deployments i.e deployments that are not reserved.

Reply:
that is why the condition below is that this number must be higher than the limit (as there is one reserved deployment) 
i.e.  6 real + 1 reserved  is still equal to capacty so test continues.

Reply:
I saw that.